src/Main.elm

```diff
@@ -11,14 +11,32 @@ import Html.Events exposing (onClick)
 
 
 type alias Model =
-    { board : List Int
-    , selectedSquare : Maybe Int
+    { selectedSquare : Maybe Int
+    , board : List BoardSquare
     }
 
 
+type alias BoardSquare =
```

Comment:
New `BoardSquare` state type. Holds the id and then the new custom type for what we talked about today regarding the state a square can be in